src/libraries/System.Net.Security/src/System/Net/Security/Pal.Android/SafeDeleteSslContext.cs

```diff
@@ -249,7 +255,13 @@ private static AsymmetricAlgorithm GetPrivateKeyAlgorithm(X509Certificate2 cert,
 
             if (!isServer && !string.IsNullOrEmpty(authOptions.TargetHost))
             {
-                Interop.AndroidCrypto.SSLStreamSetTargetHost(handle, authOptions.TargetHost);
+                // the Java SNIHostName class that's used internally to wrap the hostname
+                // doesn't support IPv6 addresses
+                var containsUnsupportedCharacters = authOptions.TargetHost.Contains(':');
+                if (!containsUnsupportedCharacters)
```

Comment:
I'm wondering if we should throw PNSP instead. This IMHO may lead to silent failures that are hard to diagnose. On the other hand, HttpClient will do it automatically and it would make it impossible to connect to IPv6 loopback. Perhaps add some diagnostic log ...?

Reply:
I removed the workaround. It doesn't belong to this PR anyway and it deserves its own dedicated issue and solution.

Reply:
https://github.com/dotnet/runtime/issues/79143